components/script/dom/fakexrdevicecontroller.rs

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::FakeXRDeviceControllerBinding::{
+    self, FakeXRDeviceControllerMethods, FakeXRRigidTransform, FakeXRViewInit,
+};
+use crate::dom::bindings::codegen::Bindings::XRViewBinding::XREye;
+use crate::dom::bindings::error::{Error, Fallible};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::globalscope::GlobalScope;
+use dom_struct::dom_struct;
+use webvr_traits::{MockVRControlMsg, WebVREyeParameters, WebVRMsg};
+
+#[dom_struct]
+pub struct FakeXRDeviceController {
+    reflector: Reflector,
+}
+
+impl FakeXRDeviceController {
+    pub fn new_inherited() -> FakeXRDeviceController {
+        FakeXRDeviceController {
+            reflector: Reflector::new(),
+        }
+    }
+
+    pub fn new(global: &GlobalScope) -> DomRoot<FakeXRDeviceController> {
+        reflect_dom_object(
+            Box::new(FakeXRDeviceController::new_inherited()),
+            global,
+            FakeXRDeviceControllerBinding::Wrap,
+        )
+    }
+
+    pub fn send_msg(&self, msg: MockVRControlMsg) {
+        self.global()
+            .as_window()
+            .webvr_thread()
+            .unwrap()
+            .send(WebVRMsg::MessageMockDisplay(msg))
+            .unwrap();
+    }
+}
+
+impl FakeXRDeviceControllerMethods for FakeXRDeviceController {
+    // check-tidy: no specs after this line
```

Comment:
No spec links :(

Reply:
there's no spec! there's https://github.com/immersive-web/webxr-test-api , but it's not really a spec. I've linked to it in a bunch of places